src/testdir/test_unlet_env.vim

```diff
@@ -0,0 +1,25 @@
+" Test for unlet $FOOBAR.
+
+func Test_UnletEnv()
+  let envcmd = has('win32') ? 'set' : 'env'
```

Comment:
And before doing that you need to check that `unlet $...` does not throw and skip the test if it does. Throwing in some environment *is* expected behaviour.